### DISCUSSION TITLE
Alpine upgrades to Edge (3.19), even though specifying 3.18

### DISCUSSION THREAD

**Discussion by tweisberger** [tteck/Proxmox#2115] [General]
I found this issue when trying to diagnose the latest release of Zigbee2MQTT breaking on Alpine. Testing by starting from scratch, I found that even specifying the version to 3.18 (which is also default), the apk repo still references edge. 
```
alpine:~# cat /etc/apk/repositories 
https://dl-cdn.alpinelinux.org/alpine/v3.18/main
https://dl-cdn.alpinelinux.org/alpine/v3.18/community
https://dl-cdn.alpinelinux.org/alpine/edge/main
https://dl-cdn.alpinelinux.org/alpine/edge/community
https://dl-cdn.alpinelinux.org/alpine/edge/testing
```
The update in the install script pushes alpine to 3.19.
```
alpine:~# cat /etc/alpine-release 
3.19_rc1
```

Similar to the base Alpine, it pushes the Alpine-z2m to edge where packages and OS is not yet stable. This wouldn't be an issue for me if z2m wasn't broken. But did cause some chaos, as I thought I was on a stable branch.

**Comment by tteck** [comment]
At the time the script was written, the Zigbee2MQTT package was only available in the edge branch, now I see that it's also available in the [3.18 branch](https://pkgs.alpinelinux.org/packages?name=zigbee2mqtt&branch=v3.18&repo=&arch=x86_64&maintainer=) but in an older version (1.31.2). The current Z2M version is 1.33.2, which is the listed version in the [edge branch](https://pkgs.alpinelinux.org/packages?name=zigbee2mqtt&branch=edge&repo=&arch=x86_64&maintainer=). 
What are you advising?

**Comment by tweisberger** [comment]
Good question. Mostly making it aware that even though a user specifies 3.18, the install will update to 3.19 before the setup scrip completes. 

Also Z2M 1.33.2 is [broken on Alpine](https://github.com/Koenkk/zigbee2mqtt/issues/19869) at the moment.
